frame/assets/src/impl_fungibles.rs

```diff
@@ -283,3 +283,12 @@ impl<T: Config<I>, I: 'static> fungibles::roles::Inspect<<T as SystemConfig>::Ac
 		Asset::<T, I>::get(asset).map(|x| x.freezer)
 	}
 }
+
+impl<T: Config<I>, I: 'static> fungibles::InspectEnumerable<T::AccountId> for Pallet<T, I> {
+	/// Returns an iterator of the assets in existence.
+	///
+	/// NOTE: iterating this list invokes a storage read per item.
+	fn asset_ids() -> Box<dyn Iterator<Item = Self::AssetId>> {
+		Box::new(Asset::<T, I>::iter_keys())
+	}
+}
```

Comment:
add `use frame_support ::storage::KeyPrefixIterator;` on top
and then it would look like this: 

```suggestion
impl<T: Config<I>, I: 'static> fungibles::InspectEnumerable<T::AccountId> for Pallet<T, I> {
	type AssetsIterator = KeyPrefixIterator<<T as Config<I>>::AssetId>;

	/// Returns an iterator of the assets in existence.
	///
	/// NOTE: iterating this list invokes a storage read per item.
	fn asset_ids() -> Self::AssetsIterator {
		Asset::<T, I>::iter_keys()
	}
}
```

Reply:
Good idea. Have updated the code.